src/Tasks/FileIO/WriteLinesToFile.cs

```diff
@@ -82,27 +83,31 @@ public override bool Execute()
 
                 try
                 {
+
                     if (Overwrite)
                     {
                         if (buffer.Length == 0)
                         {
-                            // if overwrite==true, and there are no lines to write,
-                            // just delete the file to leave everything tidy.
-                            System.IO.File.Delete(File.ItemSpec);
+                            // File.Delete throws when the directory does not exist.
+                            if (System.IO.Directory.Exists(GetTargetDirectory()))
+                            {
+                                // if overwrite==true, and there are no lines to write,
+                                // just delete the file to leave everything tidy.
+                                System.IO.File.Delete(File.ItemSpec);
+                            }
                         }
                         else
                         {
-                            string contentsAsString = null;
-
+                            CreateTargetDirectory();
+                            string contentsAsString = buffer.ToString();
```

Comment:
Removed the if below, since `buffer.ToString()` is always called in this `else` branch.